### PR TITLE
Add deck mode to tide and plank.

### PR DESCRIPTION
**Pull request by spxtr**
This lets one run these programs locally, but using the production data that is served up by deck.

If this pattern is useful for other controllers then it's pretty easy to add it to them. I need it in order to make sure that tide will work properly for k/k.

/area prow

### CODE REVIEW

File: prow/kube/client.go [hunk 1]
```diff
@@ -366,23 +372,29 @@ func (c *Client) GetProwJob(name string) (ProwJob, error) {
...
+			query: map[string]string{"labelSelector": selector},
+		}, &jl)
+		return jl.Items, err
 	}
+	var jl []ProwJob
 	err := c.request(&request{
-		method: http.MethodGet,
-		path:   fmt.Sprintf("/apis/prow.k8s.io/v1/namespaces/%s/prowjobs", c.namespace),
-		query:  map[string]string{"labelSelector": selector},
+		deckPath: "/prowjobs.js",
```
Comment (kargakis) [inline prow/kube/client.go]: Can't you append this inside `doRequest` if `deckURL` is specified?

File: prow/kube/client.go [hunk 2]
```diff
@@ -324,8 +332,7 @@ func (c *Client) GetPod(name string) (Pod, error) {
 	c.log("GetPod", name)
 	var retPod Pod
 	err := c.request(&request{
-		method: http.MethodGet,
```
Comment (cjwagner) [inline prow/kube/client.go]: Why did you remove `method: http.MethodGet` throughout? Does the method default to GET somewhere?

File: prow/kube/client.go [hunk 3]
```diff
@@ -181,8 +183,11 @@ func (c *Client) requestRetry(r *request) ([]byte, error) {
 	return rb, nil
 }
 
-func (c *Client) doRequest(method, urlPath string, query map[string]string, body interface{}) (*http.Response, error) {
+func (c *Client) doRequest(method, deckPath, urlPath string, query map[string]string, body interface{}) (*http.Response, error) {
 	url := c.baseURL + urlPath
+	if c.deckURL != "" {
```
Comment (cjwagner) [inline prow/kube/client.go]: I think this should be `if c.deckURL != "" && deckPath != "" {`
Otherwise if plank specifies a `deckURL` it won't be able to update or create prowjobs.

File: prow/kube/client.go (hunk 2 above)
Comment (spxtr) [reply prow/kube/client.go]: Yes, that's the default in go's http package.

**Comment by spxtr** [comment]
> The interaction between and the semantic meaning of the local, dryRun, and deckURL flags is somewhat confusing.

Agreed. I'll try to clean that up.

File: prow/kube/client.go (hunk 1 above)
Comment (spxtr) [reply prow/kube/client.go]: FIxed. I had it that old way because k8s says `{"items":[]}` but deck was just saying `[]`. I fixed deck to match k8s.

File: prow/kube/client.go (hunk 3 above)
Comment (spxtr) [reply prow/kube/client.go]: Done.

**Comment by spxtr** [comment]
This is about all I want to do in this PR. I opened #5665 to track further work.

File: prow/cmd/plank/main.go [hunk 4]
```diff
@@ -39,12 +39,14 @@ var (
 	totURL = flag.String("tot-url", "", "Tot URL")
 
 	configPath   = flag.String("config-path", "/etc/config/config", "Path to config.yaml.")
+	cluster      = flag.String("cluster", "", "Path to kube.Cluster YAML file. If empty, uses the local cluster.")
```
Comment (kargakis) [inline prow/cmd/plank/main.go]: We may want to sync [`--build-cluster`](https://github.com/kubernetes/test-infra/blob/1e5e44f72851b6a3e5d9db1dcb1cd76c40a19f8a/prow/cmd/deck/main.go#L45) with this in a follow-up. (or the other way around?)

Comment (spxtr) [reply prow/cmd/plank/main.go]: Yes.

**Comment by spxtr** [comment]
@cjwagner sorry to bother but could you redeploy deck with this change?

**Comment by BenTheElder** [comment]
> @cjwagner sorry to bother but could you redeploy deck with this change?

Not cjwagner, but done :-)

**Comment by BenTheElder** [comment]
`/prowjobs.js` is massive at 30.3 MB (!), it took 1 minute for content download 😱

**Comment by spxtr** [comment]
Yep! We'll probably want to do something about that at some point, but for now it's just a big list.

Thanks :)
